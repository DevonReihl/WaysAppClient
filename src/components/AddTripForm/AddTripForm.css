.addTripForm {
  display: flex;
  flex-direction: column;
  padding: 5px;
  width: 98%;
  margin: auto;
}

.addTripButton:hover {
  cursor: pointer;
}

.addTripForm > label,
.placeSearch {
  font-size: 20px;
  font-weight: bold;
}

.addTripForm > input {
  margin-bottom: 12px;
}

.addTripSection > div > h2 {
  font-size: 48px;
  font-weight: bold;
  color: rgb(46, 140, 216);
  text-align: center;
  font-family: 'Fredoka One', Arial, Helvetica, sans-serif;
}

.addTripForm > input::-webkit-input-placeholder,
.location-box::-webkit-input-placeholder {
  color: rgb(184, 184, 184);
}

#days,
#rating {
  text-align: center;
  width: 120px;
  margin-left: 10px;
}

.trip-rating-days {
  font-size: 20px;
  font-weight: bold;
  text-align: center;
}

.TripCardDemo {
  margin-top: 40px;
  margin-bottom: 40px;
}

.demo-middle {
  word-wrap: break-word;
}

.TripCardDemo-wrapper {
  display: flex;
  justify-content: center;
  margin: auto;
}

.TripCardDemo-wrapper > img {
  width: 50px;
  height: 50px;
  align-self: flex-start;
  margin: 100px 5px 0px 5px;
}

.TripCardDemo-wrapper > img:active {
  margin-top: 101px;
}

.country-menu {
  box-sizing: border-box;
  border: 2px solid #ccc;
  border-radius: 4px;
  font-size: 16px;
  background-color: rgb(255, 255, 255);
  padding: 9px 20px;
  font-weight: bold;
  width: 165px;
  margin-top: 12px;
  color: rgb(82, 82, 82);
  text-transform: lowercase;
  margin-left: 10px;
}

.country-menu > option {
  text-transform: lowercase;
}

.selectCountry {
  text-align: center;
}

.placeSearch {
  display: flex;
  justify-content: space-between;
  flex-direction: column;
}

.style-list {
  cursor: pointer;
  list-style: none;
  border-bottom: 1px solid #aaa;
  height: 40px;
  padding-left: 20px;
  line-height: 40px;
  white-space: nowrap;
  overflow: hidden !important;
  text-overflow: ellipsis;
  width: 100%;
  font-size: 14px;
}

.style-list:hover {
  background-color: #f6f6f6;
}

.TripCard-bottom-demo {
  height: 80px;
  width: 315px;
  color: rgb(8, 8, 8);

  font-family: 'Fredoka One', Arial, Helvetica, sans-serif;
  font-size: 20px;
}

.TripCard-bottom-info-demo {
  padding-top: 25px;
  width: 315px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

@media all and (min-width: 600px) {
  .addTripForm {
    width: 600px;
    margin: auto;
  }
  .addTripForm > input {
    margin-bottom: 12px;
  }

  .country-menu {
    width: 165px;
    margin-top: 0px;
    margin-left: 0px;
  }

  .selectCountry {
    text-align: left;
    margin-left: 15px;
  }

  .placeSearch {
    flex-direction: row;
  }

  .location-box {
    width: 400px;
  }

  .style-list {
    width: 398px;
  }
}

@media all and (max-width: 440px) {
  .TripCardDemo-wrapper > img {
    width: 30px;
    height: 30px;
  }

  .TripCardDemo {
    width: 295px;
  }

  .TripCard-bottom-demo {
    width: 100%;
  }

  .TripCard-bottom-info-demo {
    width: 100%;
    display: inline-block;
  }
}
