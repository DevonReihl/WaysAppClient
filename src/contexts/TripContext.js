import React from 'react';
import UserContext from './UserContext';

const TripContext = React.createContext({
  trips: [],
  currTripId: null,
  setCurrTripId: () => {},
  setTrips: () => {},
  verifyAuth: () => {},
  setLoading: () => {},
  returnUserID: () => {},
  setTripRating: () => {},
  loading: null,
});

export default TripContext;

export class TripProvider extends React.Component {
  state = {
    trips: [],
    currTripId: null,
    currPlace: {},
    stops: [],
    loading: false,
  };

  static contextType = UserContext;

  setTrips = (res) => {
    this.setState({
      trips: res,
    });
  };

  setTripRating = (trip_id) => {
    const index = this.state.trips.findIndex((trip) => {
      return trip.id === Number(trip_id);
    });
    let trips = [...this.state.trips];
    trips[index] = {
      ...trips[index],
      rating: Number(trips[index].rating) + 1,
    };
    this.setState({ trips });
  };

  setLoading = (boolean) => {
    this.setState({
      loading: boolean,
    });
  };

  setCurrTripId = (id) => {
    this.setState({
      currTripId: id,
    });
  };

  handleAddTrip = (trip) => {
    this.setState({
      trips: [...this.state.trips, trip],
    });
  };

  verifyAuth = (id) => {
    return id === this.context.user.id;
  };

  returnUserID = () => {
    return this.context.user.id;
  };

  render() {
    const value = {
      trips: this.state.trips,
      currTripId: this.state.currTripId,
      setTrips: this.setTrips,
      setCurrTripId: this.setCurrTripId,
      verifyAuth: this.verifyAuth,
      loading: this.state.loading,
      setLoading: this.setLoading,
      returnUserID: this.returnUserID,
      setTripRating: this.setTripRating,
    };

    return (
      <TripContext.Provider value={value}>
        {this.props.children}
      </TripContext.Provider>
    );
  }
}
