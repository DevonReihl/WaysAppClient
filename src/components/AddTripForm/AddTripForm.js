import React from 'react';
import TripService from '../../services/trip-service';
import TripContext from '../../contexts/TripContext';
import images from '../../assets/images/images';
import PlaceSearch from '../PlaceSearch/PlaceSearch';
import './AddTripForm.css';
import SelectCountries from './SelectCountries/SelectCountries';

export default class AddTripForm extends React.Component {
  state = {
    short_description: 'Times Square',
    activities: 'Shopping',
    days: 2,
    destination: 'New York, NY',
    place: { place: 'New York', coordinates: { lng: null, lat: null } },
    countryCode: 'us',
    error: null,
    images: [
      'city',
      'nightlife',
      'hiking',
      'mountain',
      'restaurant',
      'countryside',
      'amusementpark',
    ],
    imagesScroll: 0,
    placesTouched: false,
  };

  static contextType = TripContext;

  handleSelectCountries = (e) => {
    const country = e.currentTarget.value;
    this.setState({
      countryCode: country.slice(-2, country.length),
      place: { place: 'New York', coordinates: { lng: null, lat: null } },
    });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const { short_description, country, days, activities } = e.target;
    this.setState({
      error: null,
      country: country.value.slice(-2, country.value.length),
    });
    if (
      !this.state.place.coordinates.lng ||
      !this.state.place.coordinates.lat
    ) {
      this.setState({ error: 'Must select a real place from dropdown list.' });
      return;
    }

    let trip = {
      destination: this.state.place.place,
      short_description: short_description.value,
      long: this.state.place.coordinates.lng,
      lat: this.state.place.coordinates.lat,
      days: days.value,
      activities: activities.value,
      img: this.state.images[this.state.imagesScroll],
    };
    let currentTrips = this.context.trips;
    TripService.postTrip(trip)
      .then((res) => {
        this.context.setTrips([res, ...currentTrips]);
        this.props.history.push('/my-trips');
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  resetPlace = () => {
    this.context.setLoading(false);
  };

  storePlace = (place) => {
    if (this.state.place.coordinates.lng || this.state.place.coordinates.lat) {
      this.context.setLoading(false);
    } else {
      this.context.setLoading(true);
    }

    this.setState({
      place: place,
    });
  };

  handleScrollRight = () => {
    if (this.state.imagesScroll === this.state.images.length - 1) {
      this.setState({ imagesScroll: 0 });
    } else {
      this.setState({ imagesScroll: this.state.imagesScroll + 1 });
    }
  };

  handleScrollLeft = () => {
    if (this.state.imagesScroll === 0) {
      this.setState({ imagesScroll: 6 });
    } else {
      this.setState({ imagesScroll: this.state.imagesScroll - 1 });
    }
  };

  shortifyDestination = (dest) => {
    dest = dest.split(',');
    return dest[0];
  };

  render() {
    return (
      <>
        <section className="addTripSection">
          <div>
            <h2>Plan your Perfect Trip</h2>
            {this.state.error && <center>{this.state.error}</center>}
            <form
              className="addTripForm"
              id="addTripForm"
              action="#"
              onSubmit={this.handleSubmit}
            >
              <div className="placeSearch">
                <div className="selectDestination">
                  <label htmlFor="location-box">Destination:</label>
                  <PlaceSearch
                    countryCode={this.state.countryCode}
                    storePlace={this.storePlace}
                    resetPlace={this.resetPlace}
                  />
                </div>
                <div className="selectCountry">
                  <SelectCountries
                    handleSelectCountries={this.handleSelectCountries}
                  />
                </div>
              </div>
              <br />
              <label htmlFor="short_description">
                Type in a short description of your trip!
              </label>
              <input
                maxLength={30}
                onChange={(e) =>
                  this.setState({ short_description: e.target.value })
                }
                placeholder={'New York Shopping, Backpacking through Europe!'}
                type="text"
                name="short_description"
                id="short_description"
                required
              />
              <br />
              <label htmlFor="activities">
                What kind of activities do you plan to do on your trip?
              </label>
              <input
                onChange={(e) => this.setState({ activities: e.target.value })}
                type="text"
                name="activities"
                id="activities"
                placeholder={'Shopping, Sight-seeing, Gnoshing...'}
                maxLength={40}
                required
              />
              <br />
              <div className="trip-rating-days">
                <label htmlFor="days">
                  How many days does your trip take to complete?
                </label>
                <input
                  onChange={(e) => this.setState({ days: e.target.value })}
                  placeholder={1}
                  type="number"
                  id="days"
                  name="days"
                  max={99}
                  min={1}
                  required
                />
              </div>
              <br />
              <div className="TripCardDemo-wrapper">
                <img
                  onClick={this.handleScrollLeft}
                  src={images.arrow_left}
                  alt="left arrow"
                />
                <div className="TripCard TripCardDemo">
                  <div>
                    <div className="TripCard-topimage">
                      <img
                        src={images[this.state.images[this.state.imagesScroll]]}
                        alt="city skyline"
                      ></img>
                    </div>
                    <div className="TripCard-middle-section demo-middle">
                      <div className="TripCard-title">
                        {this.state.place.place.length > 32 ? (
                          <h2>
                            {this.shortifyDestination(this.state.place.place)}
                          </h2>
                        ) : (
                          <h2>{this.state.place.place}</h2>
                        )}

                        <span>{this.state.short_description}</span>
                      </div>
                      <div className="Activities">
                        <p>
                          Activities: <br />
                          {this.state.activities}
                        </p>
                      </div>
                    </div>
                    <div className={`TripCard-bottom-demo blue`}>
                      <div className="TripCard-bottom-info-demo">
                        Days {this.state.days} | Rating <span>N\A</span>
                      </div>
                    </div>
                  </div>
                </div>
                <img
                  onClick={this.handleScrollRight}
                  src={images.arrow_right}
                  alt="right arrow"
                />
              </div>
              <br />
              <div className="button-wrapper">
                <button className="myButton addTripButton" type="submit">
                  Submit!
                </button>
              </div>
            </form>
          </div>
        </section>
      </>
    );
  }
}
