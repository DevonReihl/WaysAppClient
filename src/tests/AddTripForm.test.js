import React from 'react';
import AddTripForm from '../components/AddTripForm/AddTripForm';
import { shallow } from 'enzyme';
import TripContext from '../contexts/TripContext';
import toJson from 'enzyme-to-json';

describe(`AddTripForm component`, () => {
  it('Renders without crashing', () => {
    shallow(
      <TripContext.Provider>
        <AddTripForm />
      </TripContext.Provider>
    );
  });
  it('Matches snapshot', () => {
    const wrapper = shallow(
      <TripContext.Provider>
        <AddTripForm />
      </TripContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
