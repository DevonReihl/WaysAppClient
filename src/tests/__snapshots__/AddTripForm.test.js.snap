// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTripForm component Matches snapshot 1`] = `<AddTripForm />`;
